share/wake/lib/core/list.wake

```diff
@@ -67,14 +67,26 @@ export def tail: List a => List a = match _
     Nil  = Nil
     _, t = t
 
-# map: create a new List by applying a function f to each element of a List.
-#
-#   map str    (seq 5) = "0", "1", "2", "3", "4", Nil
-#   map (_+10) (seq 5) = 10, 11, 12, 13, 14, Nil
-export def map (f: a => b): List a => List b =
-    def loop = match _
+# Create a new List by applying the function *mapFn* to each element of *list*.
+# The *map* function (along with *foldl*) is generally how one implements loops in wake.
+# This function (like most in wake) runs *mapFn* in parallel.
+#
+# Parameters:
+#  - *mapFn*: The function to apply to each element
+#  - *list*: The List of elements to feed to *mapFn*
+#  - returns: A new List with the outputs of *mapFn*
+#
+# Guarantees:
+#  - The resultant List has the same length as *list*
+#
+# ```
+# map str    (seq 5) = "0", "1", "2", "3", "4", Nil
```

Comment:
Maybe a bit of a nit that is specific to this function's description, but I'm wondering if the examples here should avoid using the `seq` function, since you'd have to understand that it returns a list counting up to the argument. If you don't know `map`, I feel like it's even less likely that you know `seq`.

Reply:
Sure.